Main Menu and Pause Menu

rmanz004 opened the pull request:
Added a main menu and pause menu to the game. From the main menu, the player can start or quit the game, as well as adjust the difficulty of the ghosts that roam around. From the pause menu, the player can resume or quit the game, as well as return to the main menu.

rmanz004 committed:
Main Menu and Pause Menu

Added a main menu and pause menu to the game. From the main menu, the player can start or quit the game, as well as adjust the difficulty of the ghosts that roam around. From the pause menu, the player can resume or quit the game, as well as return to the main menu.

randymester commented on Assets/Scripts/GameEnding.cs:
```diff
@@ -13,47 +13,47 @@ public class GameEnding : MonoBehaviour
+    bool IsPlayerAtExit;
+    bool IsPlayerCaught;
+    float Timer;
+    bool HasAudioPlayed;
```
These should be camelCase, not PascalCase. Also they should have explicit accessors declared at the beginning (public, private, etc) so they're easier to understand at a glance

randymester commented on Assets/Scripts/GameEnding.cs:
```diff
@@ -13,47 +13,47 @@ public class GameEnding : MonoBehaviour
+    float Timer;
+    bool HasAudioPlayed;
 
     void OnTriggerEnter(Collider other)
```
Include accessor

randymester commented on Assets/Scripts/GameEnding.cs:
```diff
@@ -13,47 +13,47 @@ public class GameEnding : MonoBehaviour
+        IsPlayerCaught = true;
     }
 
     void Update()
```
Include accessor

randymester commented on Assets/Scripts/GameEnding.cs:
```diff
@@ -13,47 +13,47 @@ public class GameEnding : MonoBehaviour
         }
     }
 
     void EndLevel(CanvasGroup imageCanvasGroup, bool doRestart, AudioSource audioSource)
```
Include accessor

randymester commented on Assets/Scripts/MainMenuManager.cs:
```diff
@@ -0,0 +1,63 @@
+	public TextMeshProUGUI currDifficulty;
+	public static float ghostSpeed = 2.0f;
+	public static float ghostAcceleration = 8f;
+	void Start()
```
Include accessor

randymester commented on Assets/Scripts/MainMenuManager.cs:
```diff
@@ -0,0 +1,63 @@
+	}
+
+	public void EasyDifficulty()
+    {
```
Fix spacing

randymester commented on Assets/Scripts/MainMenuManager.cs:
```diff
@@ -0,0 +1,63 @@
+		ghostAcceleration = 6f;
+		currDifficulty.text = "Current Difficulty:\n Easy";
+		Debug.Log("Easy selected");
+    }
```
Fix spacing

randymester commented on Assets/Scripts/MainMenuManager.cs:
```diff
@@ -0,0 +1,63 @@
+    {
+		ghostSpeed = 1.5f;
+		ghostAcceleration = 6f;
+		currDifficulty.text = "Current Difficulty:\n Easy";
```
You don't need that space after the \n

randymester commented on Assets/Scripts/MainMenuManager.cs:
```diff
@@ -0,0 +1,63 @@
+	{
+		ghostSpeed = 2.0f;
+		ghostAcceleration = 8f;
+		currDifficulty.text = "Current Difficulty:\n Medium";
```
You don't need that space after the \n

randymester commented on Assets/Scripts/MainMenuManager.cs:
```diff
@@ -0,0 +1,63 @@
+	{
+		ghostSpeed = 3.0f;
+		ghostAcceleration = 10f;
+		currDifficulty.text = "Current Difficulty:\n Hard";
```
You don't need that space after the \n

randymester commented on Assets/Scripts/Observer.cs:
```diff
@@ -7,27 +7,27 @@ public class Observer : MonoBehaviour
     public GameEnding gameEnding;
 
-    bool m_IsPlayerInRange;
+    bool IsPlayerInRange;
```
Make is camelCase, add accessor

randymester commented on Assets/Scripts/Observer.cs:
```diff
@@ -7,27 +7,27 @@ public class Observer : MonoBehaviour
-    bool m_IsPlayerInRange;
+    bool IsPlayerInRange;
 
     void OnTriggerEnter(Collider other)
```
Include accessor

randymester commented on Assets/Scripts/Observer.cs:
```diff
@@ -7,27 +7,27 @@ public class Observer : MonoBehaviour
         }
     }
 
     void OnTriggerExit(Collider other)
```
Include accessor

randymester commented on Assets/Scripts/Observer.cs:
```diff
@@ -7,27 +7,27 @@ public class Observer : MonoBehaviour
         }
     }
 
     void Update()
```
Include accessor

randymester commented on Assets/Scripts/PauseMenu.cs:
```diff
@@ -0,0 +1,47 @@
+{
+    public static bool gamePaused = false;
+    public GameObject PauseMenuUI;
+    void Update()
```
Include accessor

randymester commented on Assets/Scripts/PauseMenu.cs:
```diff
@@ -0,0 +1,47 @@
+public class PauseMenu : MonoBehaviour
+{
+    public static bool gamePaused = false;
+    public GameObject PauseMenuUI;
```
Should be camelCase

randymester commented on Assets/Scripts/PlayerMovement.cs:
```diff
@@ -6,18 +6,18 @@ public class PlayerMovement : MonoBehaviour
+    Rigidbody Rigidbody;
+    AudioSource AudioSource;
+    Vector3 Movement;
+    Quaternion Rotation = Quaternion.identity;
```
Make them camelCase and add accessors

randymester reviewed (changes requested):
Additional comments:

1. In the Rules text on the main menu, there needs to be a space between "by the"
2. Maybe add button highlighting in the main menu
3. I would like to see a button appear on the lose and win screens that brings you back to the main menu

rmanz004 committed:
Updated Win and Lose Screens

Updated the win and lose screens so that the player can either return to the main menu or exit the game when they win or lose. Tweaked other scripts as well.
